### PR TITLE
I propose this as an alternative

### PR DESCRIPTION
**Pull request by yfarjoun**
to @tfenne 's PR #194

The difference is that the change is done in collectMultipleMetrics and tests are included for 
greater good!

Comments welcome: @jacarey @ktibbett

### CODE REVIEW

**Comment by tfenne** [comment]
Looks good to me!  I almost went down that path, but when I found that makeInstance() method didn't pass through enough, I didn't want to go changing that without more testing.  I think this is the better approach - will close my PR.

**Comment by yfarjoun** [comment]
Thanks for your input @tfenne. can I consider that a code review, @nh13 ?

File: src/tests/java/picard/analysis/CollectMultipleMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                "PROGRAM=null",
+                "PROGRAM="+CollectMultipleMetrics.Program.CollectAlignmentSummaryMetrics.name(),
+                "PROGRAM="+CollectMultipleMetrics.Program.CollectInsertSizeMetrics.name()
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+
+        final MetricsFile<AlignmentSummaryMetrics, Comparable<?>> output = new MetricsFile<AlignmentSummaryMetrics, Comparable<?>>();
+        output.read(new FileReader(outfile));
+
+        for (final AlignmentSummaryMetrics metrics : output.getMetrics()) {
```
Comment (nh13) [inline src/tests/java/picard/analysis/CollectMultipleMetricsTest.java]: duplicate code in your tests.

Comment (yfarjoun) [inline src/tests/java/picard/analysis/CollectMultipleMetricsTest.java]: not really. the input files are different. it's true that I run the same two programs in both tests, but I only look at the result of the relevant one in each. the reason I run two programs is to make sure that running two programs doesn't somehow change the results.

Comment (nh13) [inline src/tests/java/picard/analysis/CollectMultipleMetricsTest.java]: I disagree. The asserts are the same, and the ```input``` and ```reference``` files are the same.

Comment (yfarjoun) [inline src/tests/java/picard/analysis/CollectMultipleMetricsTest.java]: We seem to be not looking at the same lines of code.

the lines of code you marked are a duplicate of what?

Comment (nh13) [inline src/tests/java/picard/analysis/CollectMultipleMetricsTest.java]: See: https://github.com/broadinstitute/picard/pull/195/files#diff-95ae0051bfea869e3e0d29ebad1bda73R47
See: https://github.com/broadinstitute/picard/pull/195/files#diff-a9a05ca5beab622293e1050cace330f9R263

**Comment by yfarjoun** [comment]
@nh13: thanks for catching that. A relic of my misunderstanding regarding what needs fixing (and thus what needs testing).

**Comment by yfarjoun** [comment]
updated commit.

**Comment by mattsooknah** [comment]
Looks good. Merge away!

**Pull request merged**